rclcpp/src/rclcpp/node.cpp

```diff
@@ -242,6 +242,39 @@ Node::get_parameters(
   return results;
 }
 
+const rclcpp::parameter::ParameterVariant
+Node::get_parameter(const std::string & name) const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  rclcpp::parameter::ParameterVariant parameter;
+
+  if (get_parameter_(name, parameter)) {
+    return parameter;
+  } else {
+    throw std::out_of_range("Parameter does not exist in map");
+  }
+}
+
+bool Node::get_parameter(const std::string & name,
+  rclcpp::parameter::ParameterVariant & parameter) const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  return get_parameter_(name, parameter);
+}
+
+bool Node::get_parameter_(const std::string & name,
+  rclcpp::parameter::ParameterVariant & parameter) const
```

Comment:
Hmm, maybe you could expand on why you don't just lock this function as @tfoote suggested? Seems to me that you could move the lock in the other three functions to this one without any issues.